subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/RecentlyUsedGradleVersions.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.cache.internal;
+
+import org.gradle.util.GradleVersion;
+
+import java.util.SortedSet;
+
+public interface RecentlyUsedGradleVersions {
+    SortedSet<GradleVersion> getRecentlyUsedGradleVersions();
```

Comment:
What's _recent_ about the returned result? Looking at the implementation and the logic, there is no timing information analyzed. It is only recent because something else takes care of deleting older folders. So I would rather have this named without the _recent_ part. Makes sense?

Reply:
And I just saw the opposite conversation happened before, maybe a better name would be `UsedGradleVersions` or `CurrentlyCachedGradleVersions`? Though that last one is not exactly true either.

Reply:
Point taken. I'll remove `recently` from the name.